### ISSUE TITLE
CUDA does not appear to be working on Fedora with switchable graphics

### ISSUE THREAD

**Issue by alexschneider** [mkiol/dsnote#58]
I believe this is a separate issue to #48 as it's a different error message for `cudaGetDeviceCount`. I have switchable graphics with an Nvidia RTX A3000 on a Dell Precision 5770 and I'm using the Flatpak Beta version.

I'm using [EnvyControl](https://github.com/bayasdev/envycontrol) and [these instructions for installing CUDA + Machine Learning](https://rpmfusion.org/Howto/CUDA). I've verified that other apps use the NVidia GPU:

```
$ glxgears -info
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
GL_RENDERER   = Mesa Intel(R) Graphics (ADL GT2)
GL_VERSION    = 4.6 (Compatibility Profile) Mesa 23.1.9
GL_VENDOR     = Intel
$ __NV_PRIME_RENDER_OFFLOAD=1 __VK_LAYER_NV_optimus=NVIDIA_only __GLX_VENDOR_LIBRARY_NAME=nvidia glxgears -info
Running synchronized to the vertical refresh.  The framerate should be
approximately the same as the monitor refresh rate.
GL_RENDERER   = NVIDIA RTX A3000 12GB Laptop GPU/PCIe/SSE2
GL_VERSION    = 4.6.0 NVIDIA 535.113.01
GL_VENDOR     = NVIDIA Corporation
```

Logs below:

```
[I] 13:46:28.980 0x7f676e20ed00 init:49 - logging to stderr enabled
[D] 13:46:28.980 0x7f676e20ed00 () - translation: "en_US"
[W] 13:46:28.980 0x7f676e20ed00 () - failed to install translation
[D] 13:46:28.980 0x7f676e20ed00 () - starting standalone app
[D] 13:46:28.980 0x7f676e20ed00 () - app: net.mkiol dsnote
[D] 13:46:28.980 0x7f676e20ed00 () - config location: "/home/alex/.var/app/net.mkiol.SpeechNote/config"
[D] 13:46:28.980 0x7f676e20ed00 () - data location: "/home/alex/.var/app/net.mkiol.SpeechNote/data/net.mkiol/dsnote"
[D] 13:46:28.980 0x7f676e20ed00 () - cache location: "/home/alex/.var/app/net.mkiol.SpeechNote/cache/net.mkiol/dsnote"
[D] 13:46:28.980 0x7f676e20ed00 () - settings file: "/home/alex/.var/app/net.mkiol.SpeechNote/config/net.mkiol/dsnote/settings.conf"
[D] 13:46:28.980 0x7f676e20ed00 () - platform: "wayland"
[D] 13:46:28.980 0x7f676e20ed00 () - available styles: ("Default", "Fusion", "Imagine", "Material", "org.kde.breeze", "org.kde.desktop", "Plasma", "Universal")
[D] 13:46:28.981 0x7f676e20ed00 () - style paths: ("/usr/lib/qml/QtQuick/Controls.2")
[D] 13:46:28.981 0x7f676e20ed00 () - switching to style: "org.kde.breeze"
[D] 13:46:29.46 0x7f676e20ed00 () - supported audio input devices:
ALSA lib ../../oss/pcm_oss.c:397:(_snd_pcm_oss_open) Cannot open device /dev/dsp
[D] 13:46:29.56 0x7f676e20ed00 () - "pulse"
[D] 13:46:29.131 0x7f676e20ed00 () - "default"
ALSA lib ../../../src/pcm/pcm_direct.c:2045:(snd1_pcm_direct_parse_open_conf) The field ipc_gid must be a valid group (create group audio)
[D] 13:46:29.136 0x7f676e20ed00 () - "front:CARD=CODEC,DEV=0"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_input.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire_4__source"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_output.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire_7__sink.monitor"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_output.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire_6__sink.monitor"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_output.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire_5__sink.monitor"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_output.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire__sink.monitor"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_output.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire_2__sink.monitor"
[D] 13:46:29.136 0x7f676e20ed00 () - "alsa_input.pci-0000_00_1f.3-platform-sof_sdw.HiFi__hw_sofsoundwire_1__source"
[D] 13:46:29.136 0x7f676e20ed00 add_cuda_devices:227 - scanning for cuda devices
[D] 13:46:29.136 0x7f676e20ed00 add_cuda_devices:236 - cuda version: driver=0, runtime=0
[W] 13:46:29.136 0x7f676e20ed00 add_cuda_devices:241 - cudaGetDeviceCount returned: 35
[D] 13:46:29.136 0x7f676e20ed00 add_opencl_devices:301 - scanning for opencl devices
[D] 13:46:29.170 0x7f676e20ed00 add_opencl_devices:318 - opencl number of platforms: 2
[D] 13:46:29.170 0x7f676e20ed00 add_opencl_devices:343 - opencl platform: 0, name=AMD Accelerated Parallel Processing, vendor=Advanced Micro Devices, Inc.
[D] 13:46:29.170 0x7f676e20ed00 add_opencl_devices:357 - opencl number of devices: 0
[D] 13:46:29.170 0x7f676e20ed00 add_opencl_devices:343 - opencl platform: 1, name=Clover, vendor=Mesa
[D] 13:46:29.170 0x7f676e20ed00 add_opencl_devices:357 - opencl number of devices: 0
[D] 13:46:29.170 0x7f676979e600 loop:56 - py executor loop started
[D] 13:46:29.180 0x7f676e20ed00 () - starting service: app-standalone
[D] 13:46:29.185 0x7f676e20ed00 () - module already unpacked: "rhvoicedata"
[D] 13:46:29.185 0x7f676e20ed00 () - module already unpacked: "rhvoiceconfig"
[D] 13:46:29.187 0x7f6732ffd600 () - config version: 41 41
[D] 13:46:29.187 0x7f6732ffd600 () - checksum ok: "674d3682" "en.tflite"
[D] 13:46:29.187 0x7f6732ffd600 () - checksum ok: "40fe6178" "en_hv.tflite"
[D] 13:46:29.187 0x7f6732ffd600 () - checksum ok: "40fe6178" "en_hv.tflite"
[D] 13:46:29.188 0x7f6732ffd600 () - checksum ok: "dfa47af8" "multilang_whisper_large.ggml"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "multilang_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - checksum ok: "6571cb18" "en_whisper_base.ggml"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "en_whisper_base"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "en_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "es_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "it_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "pt_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "de_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "ja_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "ru_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "nl_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "id_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "ca_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "fr_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "tr_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "sv_whisper_large"
[D] 13:46:29.188 0x7f6732ffd600 () - found model: "uk_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ms_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "no_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "fi_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "cs_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "sk_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "zh_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "vi_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "th_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "el_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "hr_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "bg_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ro_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "sl_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "bs_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "mk_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "sr_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "hu_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "hi_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "fa_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "da_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "et_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "eu_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "is_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ka_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "kk_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ko_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "lt_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "lv_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "mn_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "mt_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ne_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "sq_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "sw_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "tl_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "tt_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "uz_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "yo_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "am_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ar_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "bn_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "pl_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "af_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "he_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "gu_whisper_large"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "ha_whisper_large"
[D] 13:46:29.189 0x7f676e20ed00 () - module already unpacked: "espeakdata"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "jv_whisper_large"
[D] 13:46:29.189 0x7f676e20ed00 () - mbrola dir: "/app/bin"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "te_whisper_large"
[D] 13:46:29.189 0x7f676e20ed00 () - espeak dir: "/app/bin"
[D] 13:46:29.189 0x7f6732ffd600 () - found model: "tn_whisper_large"
[D] 13:46:29.189 0x7f676e20ed00 () - default stt model not found: "en_whisper_large"
[D] 13:46:29.189 0x7f676e20ed00 () - default tts model not found: "en_coqui_vctk_vits_p376"
[D] 13:46:29.189 0x7f676e20ed00 () - default mnt lang not found: "en"
[D] 13:46:29.189 0x7f676e20ed00 () - new default mnt lang: "en"
[D] 13:46:29.189 0x7f676e20ed00 () - service refresh status, new state: busy
[D] 13:46:29.189 0x7f676e20ed00 () - service state changed: unknown => busy
[D] 13:46:29.189 0x7f676e20ed00 () - delaying features availability
[D] 13:46:29.190 0x7f6732ffd600 () - checksum ok: "85a3eee1" "en_vosk_large"
[D] 13:46:29.190 0x7f6732ffd600 () - found model: "en_vosk_large"
[D] 13:46:29.191 0x7f6732ffd600 () - checksum ok: "d8ee9054" "en_vosk_indian_large"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "en_vosk_indian_large"
[D] 13:46:29.191 0x7f6732ffd600 () - checksum ok: "b8b03e8a" "multilang_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "multilang_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "en_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "de_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "fr_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "es_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "it_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "pl_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "nl_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "cs_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "pt_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "sl_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "el_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "sv_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "hu_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "ro_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "sk_hftc_kredor"
[D] 13:46:29.191 0x7f6732ffd600 () - found model: "da_hftc_kredor"
[D] 13:46:29.195 0x7f6732ffd600 () - checksum ok: "4fb49943" "en_piper_us_ryan_high"
[D] 13:46:29.195 0x7f6732ffd600 () - found model: "en_piper_us_ryan_high"
[D] 13:46:29.195 0x7f6732ffd600 () - checksum ok: "3d46ac40" "en_piper_us_ryan_medium"
[D] 13:46:29.195 0x7f6732ffd600 () - found model: "en_piper_us_ryan_medium"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "am_espeak_am"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "bg_espeak_bg"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "bs_espeak_bs"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ca_espeak_ca"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "cs_espeak_cs"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "da_espeak_da"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "de_espeak_de"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "el_espeak_el"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "en_espeak_en"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "eo_espeak_eo"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "es_espeak_es"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "et_espeak_et"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "eu_espeak_eu"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "is_espeak_is"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "fa_espeak_fa"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "fi_espeak_fi"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "fr_espeak_fr"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "hi_espeak_hi"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "hr_espeak_hr"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "hu_espeak_hu"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "id_espeak_id"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "it_espeak_it"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ja_espeak_ja"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "kk_espeak_kk"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ko_espeak_ko"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "lv_espeak_lv"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "lt_espeak_lt"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "mk_espeak_mk"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ms_espeak_ms"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ne_espeak_ne"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "nl_espeak_nl"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "no_espeak_no"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "pt_espeak_pt"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "pt_espeak_pt_br"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ro_espeak_ro"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ru_espeak_ru"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "sk_espeak_sk"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "sl_espeak_sl"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "sr_espeak_sr"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "sv_espeak_sv"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "sw_espeak_sw"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "th_espeak_th"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "tr_espeak_tr"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "uk_espeak_uk"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ka_espeak_ka"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ky_espeak_ky"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "la_espeak_la"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "tt_espeak_tt"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "sq_espeak_sq"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "uz_espeak_uz"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "vi_espeak_vi"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "zh_espeak_yue"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "zh_espeak_hak"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "zh_espeak_cmn"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "ga_espeak_ga"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "mt_espeak_mt"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "bn_espeak_bn"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "pl_espeak_pl"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "af_espeak_af"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "he_espeak_he"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "gu_espeak_gu"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "te_espeak_te"
[D] 13:46:29.196 0x7f6732ffd600 () - found model: "tn_espeak_tn"
[D] 13:46:29.198 0x7f6732ffd600 () - checksum ok: "5fce07ba" "en_rhvoice_alan"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_rhvoice_alan"
[D] 13:46:29.198 0x7f6732ffd600 () - checksum ok: "9cdb9401" "en_coqui_vctk_vits"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_ed"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p225"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p226"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p227"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p228"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p229"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p230"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p300"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p301"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p302"
[D] 13:46:29.198 0x7f6732ffd600 () - found model: "en_coqui_vctk_vits_p376"
[D] 13:46:29.199 0x7f6732ffd600 () - checksum ok: "8fe5fa3a" "en_coqui_fairseq_eng"
[D] 13:46:29.199 0x7f6732ffd600 () - found model: "en_coqui_fairseq_eng"
[D] 13:46:29.202 0x7f6732ffd600 () - models changed
[D] 13:46:29.375 0x7f676e20ed00 () - starting app: app-standalone
[D] 13:46:29.375 0x7f676e20ed00 () - app service state: unknown => busy
[W] 13:46:29.375 0x7f676e20ed00 () - hot keys are supported only under x11
[W] 13:46:29.388 0x7f676e20ed00 ():36 - file:///usr/lib/qml/QtQuick/Controls.2/org.kde.breeze/ScrollView.qml:36:25: QML ScrollBar: Binding loop detected for property "x"
[W] 13:46:29.391 0x7f676e20ed00 ():36 - file:///usr/lib/qml/QtQuick/Controls.2/org.kde.breeze/ScrollView.qml:36:25: QML ScrollBar: Binding loop detected for property "x"
[W] 13:46:29.393 0x7f676e20ed00 ():36 - file:///usr/lib/qml/QtQuick/Controls.2/org.kde.breeze/ScrollView.qml:36:25: QML ScrollBar: Binding loop detected for property "x"
logger error: invalid format string
qrc:/qml/main.qml:267:5: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
logger error: invalid format string
qrc:/qml/main.qml:258:5: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
[W] 13:46:29.407 0x7f676e20ed00 virtual QVariant ModelSource::item(int) const:81 - ModelSource: Invalid role  -1 "color"
[W] 13:46:29.407 0x7f676e20ed00 virtual QVariant ModelSource::item(int) const:81 - ModelSource: Invalid role  -1 "color"
logger error: invalid format string
qrc:/qml/Notepad.qml:24:5: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
logger error: invalid format string
qrc:/qml/Translator.qml:29:5: QML Connections: Implicitly defined onFoo properties in Connections are deprecated. Use this syntax instead: function onFoo(<arguments>) { ... }
[W] 13:46:29.533 0x7f676e20ed00 ():144 - qrc:/qml/MainToolBar.qml:144:21: QML MenuItem: Binding loop detected for property "__reserveSpaceForIcon"
[W] 13:46:29.533 0x7f676e20ed00 ():144 - qrc:/qml/MainToolBar.qml:144:21: QML MenuItem: Binding loop detected for property "__reserveSpaceForIcon"
[W] 13:46:29.535 0x7f676e20ed00 ():46 - qrc:/qml/MainToolBar.qml:46:21: QML MenuItem: Binding loop detected for property "__reserveSpaceForIcon"
[W] 13:46:29.535 0x7f676e20ed00 ():46 - qrc:/qml/MainToolBar.qml:46:21: QML MenuItem: Binding loop detected for property "__reserveSpaceForIcon"
[D] 13:46:29.537 0x7f676e20ed00 onCompleted:153 - default font pixel size: 12
[D] 13:46:29.542 0x7f676e20ed00 () - default mnt lang not found: "en"
[D] 13:46:29.542 0x7f676e20ed00 () - new default mnt lang: "en"
[D] 13:46:29.542 0x7f676e20ed00 () - service refresh status, new state: idle
[D] 13:46:29.542 0x7f676e20ed00 () - service state changed: busy => idle
[D] 13:46:29.542 0x7f676e20ed00 () - service refresh status, new state: idle
[D] 13:46:29.599 0x7f676e20ed00 () - app service state: busy => idle
[D] 13:46:29.599 0x7f676e20ed00 () - app stt available models: 0 => 4
[D] 13:46:29.600 0x7f676e20ed00 () - update listen
[D] 13:46:29.601 0x7f676e20ed00 () - app active stt model: "" => "en_whisper_large"
[D] 13:46:29.601 0x7f676e20ed00 () - update listen
[D] 13:46:29.601 0x7f676e20ed00 () - app tts available models: 0 => 11
[D] 13:46:29.601 0x7f676e20ed00 () - update listen
[D] 13:46:29.601 0x7f676e20ed00 () - app active tts model: "" => "en_coqui_vctk_vits_p376"
[D] 13:46:29.601 0x7f676e20ed00 () - update listen
[W] 13:46:29.603 0x7f676e20ed00 () - no available mnt langs
[W] 13:46:29.603 0x7f676e20ed00 () - no available mnt out langs
[W] 13:46:29.603 0x7f676e20ed00 () - no available tts models for in mnt
[W] 13:46:29.603 0x7f676e20ed00 () - no available tts models for out mnt
[W] 13:46:29.603 0x7f676e20ed00 () - invalid task, reseting task state
[D] 13:46:29.603 0x7f676e20ed00 () - app stt configured: false => true
[D] 13:46:29.604 0x7f676e20ed00 () - app tts configured: false => true
[D] 13:46:29.605 0x7f676e20ed00 () - app busy: true => false
[D] 13:46:29.605 0x7f676e20ed00 () - stt models changed
[D] 13:46:29.605 0x7f676e20ed00 () - update listen
[D] 13:46:29.605 0x7f676e20ed00 () - tts models changed
[D] 13:46:29.605 0x7f676e20ed00 () - update listen
[W] 13:46:29.605 0x7f676e20ed00 () - no available tts models for in mnt
[W] 13:46:29.605 0x7f676e20ed00 () - no available tts models for out mnt
[D] 13:46:29.605 0x7f676e20ed00 () - ttt models changed
[D] 13:46:29.605 0x7f676e20ed00 () - app ttt configured: false => true
[D] 13:46:29.607 0x7f676e20ed00 () - mnt langs changed
[D] 13:46:29.607 0x7f676e20ed00 () - update listen
[W] 13:46:29.607 0x7f676e20ed00 () - no available mnt langs
[W] 13:46:29.607 0x7f676e20ed00 () - no available mnt out langs
[D] 13:46:30.342 0x7f676e20ed00 () - trying features availability update: false
[D] 13:46:31.317 0x7f676e20ed00 () - trying features availability update: false
[D] 13:46:32.344 0x7f676e20ed00 () - trying features availability update: false
[D] 13:46:33.342 0x7f676e20ed00 () - trying features availability update: false
[D] 13:46:33.938 0x7f676e20ed00 () - exiting
```

**Comment by mkiol** [comment]
Thanks for the report.

> I'm using [EnvyControl](https://github.com/bayasdev/envycontrol) and [these instructions for installing CUDA + Machine Learning](https://rpmfusion.org/Howto/CUDA).

Actually, local CUDA installation doesn't really matter. Speech Note flatpak package comes with all needed CUDA libraries. This is an advantage - you don't need to install anything else - and also disadvantage because package size is huge.

This error:

```
[D] 13:46:29.136 0x7f676e20ed00 add_cuda_devices:227 - scanning for cuda devices
[D] 13:46:29.136 0x7f676e20ed00 add_cuda_devices:236 - cuda version: driver=0, runtime=0
[W] 13:46:29.136 0x7f676e20ed00 add_cuda_devices:241 - cudaGetDeviceCount returned: 35
```

indicates that NVIDIA card was not detected in your system. I see the same log on my computer with AMD card. 

I have to admit I know nothing about PRIME technology and GPU switching on laptops. Most likely I won't tell you anything you already know. NVIDIA card needs to be somehow turn-on before starting an app. As I understand 'EnvyControl' should make this happen.

The Internet mentions also that there is `prime-run` tool which switches GPU by setting the following env variables: 

```
__NV_PRIME_RENDER_OFFLOAD=1
__GLX_VENDOR_LIBRARY_NAME=nvidia
```

so, you might try to do:

```
__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia flatpak run net.mkiol.SpeechNote --verbose
```

or set these variables inside flatpak sandbox using `flatpak override`:

```
flatpak override --env=__NV_PRIME_RENDER_OFFLOAD=1 --env=__GLX_VENDOR_LIBRARY_NAME=nvidia net.mkiol.SpeechNote

flatpak run net.mkiol.SpeechNote --verbose
```

Please have in mind that I'm just guessing.

**Comment by mkiol** [comment]
Maybe this guide: https://askubuntu.com/questions/1316566/how-do-you-offload-a-flatpak-application-to-run-on-the-nvidia-graphics-card-when

**Comment by alexschneider** [comment]
Thanks!

I had tried `__NV_PRIME_RENDER_OFFLOAD=1 __GLX_VENDOR_LIBRARY_NAME=nvidia flatpak run net.mkiol.SpeechNote --verbose` and it wasn't working, but your instructions for setting the variables in the flatpak sandbox using flatpak override did actually seem to fix the issue. Really appreciate it!

**Comment by mkiol** [comment]
Wow! That is awesome!

**Comment by mkiol** [comment]
Just FYI. Yesterday I have pushed Beta8 to `flathub-beta`. It brings CUDA support for Faster-Whisper STT and Coqui TTS.
